Add app isolation details from customer feedback (#23068)
dotnet/AspNetCore.Docs: aspnetcore/security/data-protection/configuration/overview.md

If the Data Protection system isn't provided by an ASP.NET Core host (for example, if you instantiate it via the `DataProtectionProvider` concrete type) app isolation is disabled by default. When app isolation is disabled, all apps backed by the same keying material can share payloads as long as they provide the appropriate [purposes](xref:security/data-protection/consumer-apis/purpose-strings). To provide app isolation in this environment, call the [SetApplicationName](#setapplicationname) method on the configuration object and provide a unique name for each app.

> Data Protection is developed with the assumption that all applications sharing a key ring can access all items in that key ring, the application unique identifier is used to ioslate application specific keys derived from the key ring provided keys. It does not expect item level permissions, such as those provided by Azure KeyVault to be used to enforce extra isolation, and this will lead to application errors. If you do not want to rely on the built-in application isolation then seperate key store locations should be used and not shared between applications.

Reviewer: fix some typos

```suggestion
Data Protection is developed with the assumption that all applications sharing a key ring can access all items in that key ring, the application unique identifier is used to isolate application specific keys derived from the key ring provided keys. It does not expect item level permissions, such as those provided by Azure KeyVault to be used to enforce extra isolation, and this will lead to application errors. If you do not want to rely on the built-in application isolation then separate key store locations should be used and not shared between applications.
```

Reviewer 2: @Haok suggestion plus a few of my own

```suggestion
Data Protection is developed with the assumption that all applications sharing a key ring can access all items in that key ring. The application unique identifier is used to isolate application specific keys derived from the key ring provided keys. It doesn't expect item level permissions, such as those provided by Azure KeyVault to be used to enforce extra isolation. Attempting item level permissions generates application errors. If you don't want to rely on the built-in application isolation, separate key store locations should be used and not shared between applications.
```
